platform/javascript/export/export_plugin.cpp

```diff
@@ -0,0 +1,671 @@
+	const String offline_page = p_preset->get("progressive_web_app/offline_page");
+	if (!offline_page.is_empty()) {
+		DirAccess *da = DirAccess::create(DirAccess::ACCESS_FILESYSTEM);
+		const String offline_dest = dir.plus_file(name + ".offline.html");
+		err = da->copy(ProjectSettings::get_singleton()->globalize_path(offline_page), offline_dest);
+		if (err != OK) {
+			EditorNode::get_singleton()->show_warning(TTR("Could not read file:") + "\n" + offline_dest);
+			return err;
+		}
+	}
+
+	// Manifest
+	const char *modes[4] = { "fullscreen", "standalone", "minimal-ui", "browser" };
+	const char *orientations[3] = { "any", "landscape", "portrait" };
+	const int display = CLAMP(int(p_preset->get("progressive_web_app/display")), 0, 4);
+	const int orientation = CLAMP(int(p_preset->get("progressive_web_app/orientation")), 0, 3);
+
+	Dictionary manifest;
+	String proj_name = ProjectSettings::get_singleton()->get_setting("application/config/name");
+	if (proj_name.is_empty()) {
+		proj_name = "Godot Game";
+	}
+	manifest["name"] = proj_name;
+	manifest["start_url"] = "./" + name + ".html";
+	manifest["display"] = String::utf8(modes[display]);
+	manifest["orientation"] = String::utf8(orientations[orientation]);
+	manifest["background_color"] = "#" + p_preset->get("progressive_web_app/background_color").operator Color().to_html(false);
+
+	Array icons_arr;
+	const String icon144_path = p_preset->get("progressive_web_app/icon_144x144");
+	err = _add_manifest_icon(p_path, icon144_path, 144, icons_arr);
+	if (err != OK) {
+		return err;
+	}
+	const String icon180_path = p_preset->get("progressive_web_app/icon_180x180");
+	err = _add_manifest_icon(p_path, icon180_path, 180, icons_arr);
+	if (err != OK) {
+		return err;
+	}
+	const String icon512_path = p_preset->get("progressive_web_app/icon_512x512");
+	err = _add_manifest_icon(p_path, icon512_path, 512, icons_arr);
+	if (err != OK) {
+		return err;
+	}
+	manifest["icons"] = icons_arr;
+
+	CharString cs = Variant(manifest).to_json_string().utf8();
+	err = _write_or_error((const uint8_t *)cs.get_data(), cs.length(), dir.plus_file(name + ".manifest.json"));
+	if (err != OK) {
+		return err;
+	}
+
+	return OK;
+}
+
+void EditorExportPlatformJavaScript::get_preset_features(const Ref<EditorExportPreset> &p_preset, List<String> *r_features) {
+	if (p_preset->get("vram_texture_compression/for_desktop")) {
+		r_features->push_back("s3tc");
+	}
+
+	if (p_preset->get("vram_texture_compression/for_mobile")) {
+		String driver = ProjectSettings::get_singleton()->get("rendering/driver/driver_name");
+		if (driver == "GLES2") {
+			r_features->push_back("etc");
+		} else if (driver == "Vulkan") {
+			// FIXME: Review if this is correct.
+			r_features->push_back("etc2");
+		}
+	}
+	ExportMode mode = (ExportMode)(int)p_preset->get("variant/export_type");
+	if (mode == EXPORT_MODE_THREADS) {
+		r_features->push_back("threads");
+	} else if (mode == EXPORT_MODE_GDNATIVE) {
+		r_features->push_back("wasm32");
+	}
```

Comment:
What's the reason to not just always have `wasm32` as an export feature for the JavaScript platform?